packages/compass-indexes/src/components/create-index-modal/create-index-modal.tsx

```diff
@@ -0,0 +1,220 @@
+import { toggleUseTtl } from '../../modules/create-index/use-ttl';
+import { ttlChanged } from '../../modules/create-index/ttl';
+import { toggleUseWildcardProjection } from '../../modules/create-index/use-wildcard-projection';
+import { toggleUseColumnstoreProjection } from '../../modules/create-index/use-columnstore-projection';
+import { wildcardProjectionChanged } from '../../modules/create-index/wildcard-projection';
+import { columnstoreProjectionChanged } from '../../modules/create-index/columnstore-projection';
+import { partialFilterExpressionChanged } from '../../modules/create-index/partial-filter-expression';
+import { toggleUseCustomCollation } from '../../modules/create-index/use-custom-collation';
+import { collationStringChanged } from '../../modules/create-index/collation-string';
+import { openLink } from '../../modules/link';
+import { createIndex } from '../../modules/create-index';
+import { resetForm } from '../../modules/reset-form';
+import type { CreateIndexProps } from '../create-index-form';
+import CreateIndexForm from '../create-index-form';
+import { toggleUseIndexName } from '../../modules/create-index/use-index-name';
+
+const { track } = createLoggerAndTelemetry('COMPASS-IMPORT-EXPORT-UI');
+
+const bannerStyles = css({
+  marginTop: spacing[3],
+});
+
+const createIndexModal = css({
+  'div[role=dialog] > :first-child': {
+    maxHeight: '80vh',
+    overflow: 'scroll',
+  },
+});
+
+/**
+ * Component for the create index modal.
+ */
+class CreateIndexModal extends PureComponent<
+  Omit<CreateIndexProps, 'darkMode'>
+> {
+  static displayName = 'CreateIndexModal';
+
+  handleShow() {
+    track('Screen', { name: 'create_index_modal' });
+  }
+
+  /**
+   * Close modal and fire clear index form action.
+   */
+  handleClose() {
+    this.props.toggleIsVisible(false);
+    this.props.resetForm();
+  }
+
+  /**
+   * Close modal and fire create index form action.
+   */
+  onConfirm = () => {
+    this.props.createIndex();
+  };
+
+  /**
+   * Close modal and fire toggle is visible form action.
+   */
+  onCancel = () => {
+    this.props.resetForm();
+    this.props.toggleIsVisible(false);
+  };
+
+  /**
+   * Render an error banner.
+   */
+  renderError() {
+    if (!this.props.error) {
+      return;
+    }
+
+    return (
+      <Banner
+        className={bannerStyles}
+        variant="danger"
+        dismissible
+        onClose={this.props.clearError}
+      >
+        {this.props.error}
+      </Banner>
+    );
+  }
+
+  /**
+   * Render in progress banner.
+   */
+  renderInProgress() {
+    if (!this.props.error || !this.props.inProgress) {
+      return;
+    }
+
+    return (
+      <Banner className={bannerStyles} variant="info" dismissible>
+        Create in Progress
+      </Banner>
+    );
+  }
+
+  /**
+   * Render the create index modal.
+   *
+   * @returns {React.Component} The create index modal.
+   */
+  render() {
+    this.handleShow();
+    return (
+      <ConfirmationModal
```

Comment:
In a few other places in Compass we use `ConfirmationModal` where we probably want to be using a regular `Modal` with our own styles. (I think I introduced it initially oops). Since we're updating this modal is this a good time to use `Modal`? If we make a footer we can use that in the ticket we have for updating the other places to use the regular `Modal`.

Here's a ticket where we're tracking that: https://jira.mongodb.org/browse/COMPASS-5522

Reply:
Replaced with Modal! Yeah, I took a look at other placed and thought we intentionally use ConfirmationModal :)